src/goodbase/services/evg_service.py

```diff
@@ -63,15 +64,42 @@ def analyze_build(build: Build) -> Optional[BuildStatus]:
             all_tasks=all_tasks,
         )
 
-    def check_version(self, evg_version: Version, build_checks: List[BuildChecks]) -> bool:
+    def _get_successful_tasks(self, tasks: List[Task], allow_known_failures: bool) -> set[str]:
```

Comment:
[nit] I know this is not a public method, but having a docstring would help us in the future and you have added it to the private method below anyway :).